.github/workflows/ci.yml

```diff
@@ -0,0 +1,89 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ci-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: Lint (ruff + mypy)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+          cache: "pip"
+
+      - name: Install dependencies for linting
+        run: |
+          python -m pip install -U pip setuptools wheel
+          python -m pip install mypy ruff pandas-stubs "numpy>=1.26.3,<=2.1.3" matplotlib xgboost pandas scikit-learn openpyxl jinja2 python-ternary pytest shap interpret types-tqdm types-Pillow scikit-image lightgbm lime opencv-python-headless==4.10.0.84 pyyaml types-pyyaml
+
+      - name: Run mypy
+        run: mypy .
+
+      - name: Run ruff
+        run: ruff check .
+
+  functional_test:
+    name: Functional test (python main.py)
```

Comment:
the functional test should probably be removed, and other comments about this file from gh-5 also hold here, I think